src/modules/keepalive/keepalive_api.c

```diff
@@ -142,3 +140,100 @@ ka_state ka_destination_state(str *destination)
 
 	return ka_dest->state;
 }
+/*!
+* @function ka_del_destination
+* @abstract deletes given sip uri in allocated destination stack as named ka_alloc_destinations_list
+*
+* @param msg sip message
+* @param uri given uri
+* @param owner given owner name, not using now
+*	*
+* @result 1 successful  , -1 fail
+*/
+int ka_del_destination(str *uri, str *owner){
+
+	ka_dest_t *target=0,*head=0;
+
+	if(!ka_find_destination(uri,owner,&target,&head)){
+		LM_ERR("Couldnt find destination \r\n");
+		return -1;
+	}
```

Comment:
Similar issues, `target` and `head` are retrieved from `ka_find_destination()`, followed by code out of the mutex zone  and later used directly, but at that time any of them can be already deleted. So removal has to be done in the same mutex zone as it is searched.

Actually I do not see any good use for  `ka_find_destination()` returning `target` and `head`. The function can be useful to know if the address exists, but using `target` and `head` after this function expose to segfault. Either you keep the lock set in the function at the return time and unlock later, out of the function, or just make it to return true/false on finding the destination, without returning `target` and `head`.